src/ngAnimate/animateRunner.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+var $$rAFMutexFactory = ['$$rAF', function($$rAF) {
+  return function() {
+    var passed = false;
+    $$rAF(function() {
+      passed = true;
+    });
+    return function(fn) {
+      passed ? fn() : $$rAF(fn);
+    };
+  }
+}];
+
+var $AnimateRunnerFactory = ['$q', '$$rAFMutex', function($q, $$rAFMutex) {
+  var DONE_PENDING_STATE = 1;
+  var DONE_COMPLETE_STATE = 2;
+
+  AnimateRunner.chain = function(chain, callback) {
+    var status = true;
+    var index = 0;
+
+    next();
+    function next() {
+      if (index === chain.length) {
+        callback(status);
+        return;
+      }
+
+      chain[index](function(response) {
+        status = status && response;
+        index++;
+        next();
+      });
+    }
+  };
+
+  AnimateRunner.all = function(runners, callback) {
+    var count = 0;
+    var status = true;
+    forEach(runners, function(runner) {
+      runner.done(onProgress);
+    });
+
+    function onProgress(response) {
+      status = status && response;
+      if (++count === runners.length) {
+        callback(status);
+      }
+    }
+  };
+
+  function AnimateRunner(host) {
+    this.setHost(host);
+
+    this._doneCallbacks = [];
+    this._atleastOneRAFPassedSinceStart = $$rAFMutex();
```

Comment:
I would just call this something like `batchCall` or `runInAnimationFrame`. The fact that it uses rAF does not need to be exposed.